slowquant/qiskit_interface/util.py

```diff
@@ -92,6 +94,60 @@ def get_bitstring_sign(op: str, binary: int) -> int:
     return 1
 
 
+def swap_indices(num_qubits: int, swap: tuple[int, int]) -> list[int]:
+    """Find new bit ordering based on a bit swap.
+
+    Args:
+        num_qubits: Number of qubits
+        swap: Swap to be performed
+
+    Returns:
+        List of new ordering in qubit basis based on swap.
+    """
+    number = 2**num_qubits
+    pos1, pos2 = swap
+
+    # original list in decimals
+    decimal_list = np.arange(number)
+
+    # Generate the binary representation matrix
+    binary_matrix = ((decimal_list[:, None] & (1 << np.arange(num_qubits)[::-1])) > 0).astype(int)
```

Comment:
We are making a matrix here that scales as `2**num_qubits * 2**num_qubits` in size? 
16 qubits would be ~35 GB.

Reply:
Yes. it scales as M. But it is also just used for M that won't exceed 8 qubits.

Reply:
That is a good point